uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/staff-leave-report.component.js

```diff
@@ -145,6 +147,16 @@ define([
         });
     }
 
+    /**
+     * Indexes the leave request data of a section and stores it in the
+     * dataIndex attribute.
+     *
+     * @param {Object} section - the section object that contains data to index.
+     */
+    function indexSectionData (section) {
+      section.dataIndex = _.indexBy(section.data, 'id');
+    }
+
     /**
      * NOTE: This should be just temporary, see PCHR-1810
```

Comment:
Is this still relevant after refactoring? Not asking for any changes but would be great to know.

Reply:
Not sure why this is "temporary", looks like ok code to me. What should have been different? The ticket doesn't seem to be related to what this function accomplishes. Yes, they both are about leave request statuses, but this one is about loading all possible status values so they can be used to query leave request types, and the one in the ticket is about determining if a specific leave request has a determined status.

Lets wait for @deb1990 to see what kind of refactoring he had in mind.